Add a reverse proxy template

goetzk opened the pull request:
After asking in #11 I decided to just publish my config to help others easily
deploy a reverse proxy.

goetzk committed:
WIP: Add a reverse proxy template

After asking in #11 I decided to just publish my config to help others easily
deploy a reverse proxy.

goetzk commented:
Outstanding issues:
- proper testing
- Where to document its usage, noting that the template becomes mandatory 
 - (might be able to change template if proxy_pass is set?)
- decide if client_max_body_size needs to be a variable
- possibly those listen directives, but the config is meant to have automatic upgrade to https so being able to customise them doesn't necessarily make sense.

franklinkim commented:
Hi, thanks for the PR.

* documentation should be placed within `default/main.yml` if possible

I would suggest adding a new top level object `nginx_proxies` with options similar to `nginx_sites`. Thus you can provide a simple working revers proxy template with the least configuration given the possibility to override it with `item.template`. Maybe add an additional `item.options` dictionary where custom configuration can go.

goetzk committed:
Continue development of proxies support

Somewhat based on feedback in PR, a number of changes.

goetzk commented:
I've expanded in that direction, but not completed the work (or performed further testing) yet.

goetzk committed:
Changes to role following testing

This has been used to deploy a site now; the following changes were made as
part of that process.

goetzk commented:
@franklinkim 
I'm hoping you can have another look at this and provide any follow up comments.

There appears to be a CI fail which I haven't looked at yet however I've just used this role to a reverse proxy so its had one successful use.

An outstanding feature might be to automatically load the proxies template if none is specified; not sure what you think about that?

franklinkim commented:
Hi, I took another look and merged / modified your PR [here](https://github.com/weareinteractive/ansible-nginx/tree/goetzk-reverse-proxy). 

Couple of things:

- your template didn't work if you're not using ssl
- made proxies template similar to sites
- made the default proxy template should be as simple as possible
- you can add additional configurations through `item.append`, `item.rules` (letsencrypt) or a custom template

Let me know if this will work for you before I release it. THX

franklinkim commented:
@goetzk were you able to find some time to check out the modified branch?
